azurerm/internal/services/network/virtual_hub_connection_resource.go

```diff
@@ -76,6 +77,80 @@ func resourceArmVirtualHubConnection() *schema.Resource {
 				ForceNew: true,
 				Default:  false,
 			},
+
+			"routing_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"associated_route_table_id": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: azure.ValidateResourceID,
+						},
+
+						"propagated_route_table": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Computed: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"labels": {
+										Type:     schema.TypeSet,
+										Optional: true,
+										Computed: true,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
```

Comment:
Seems we cannot validate it since api document and azure portal and api returned error message don't show the validation rule after do more testing. And it allows empty string.

Reply:
Can you at least validate its not an empty string, and what happens if you use special chars `$%^&*(`? no error?

Reply:
Based on previous testing, seems API can accept any value for this property. OK. I added empty string validation.